clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1965,6 +1966,15 @@ public void processConsumerRecord(
       // update the offset metadata in the OffsetRecord
       updateOffsetMetadataInOffsetRecord(partitionConsumptionState);
       syncOffset(kafkaVersionTopic, partitionConsumptionState);
+      if (isEOPReceivedBefore == false && offsetRecord.isEndOfPushReceived() == true) {
```

Comment:
Can we add some resilience here?
The Server could fail after syncing the offset triggered by processing EOP and before deleting/in the middle of deleting sst files, right?

So it will be good to add some resilience here:
1. At startup time, if the checkpointed state is `completed` or after `EOP`, always trigger this cleanup function.

What do you think?

Reply:
added changes in the latest diff:
If SN crashes after syncing EOP but before completely deleting the sst files, it will be handled during the SN restart in `RocksDBSstFileWriter#open` via `RocksDBSstFileWriter#removeSSTFilesAfterCheckpointing`.

Reply:
`RocksDBSstFileWriter#open` will only be invoked in `beginBatchWrite` function of `RocksDBStoragePartition` and `ReplicationMetadataRocksDBStoragePartition`, so it won't be able to handle the cleanup task of the above situation, right?
If the node get restarted after syncing `EOP` and before deleting temp sst files, the next restart won't process `EOP` again, of coz, not `SOP`, so `beginBatchWrite` won't be invoked.

Reply:
Thanks for catching this. looks like `deferredWrite` controls this flow which is set based on EOP received from the Offset Record, so in this case won't end up calling `RocksDBSstFileWriter#open`. 
Calling `cleanupAfterEndOfPush()` in `checkConsumptionStateWhenStart` as well to add this resiliency.